### PR TITLE
The label for attribute should point to the input control elements id

### PR DESCRIPTION
**Pull request by ferdy-lw**
The Forms Field view label element 'for' attribute should reference the id of the form control element.

### CODE REVIEW

**Comment by ferdy-lw** [comment]
Looks like this is a dup of #24...

**Comment by dmathieu** [comment]
@bradleypriest: what do you think of this ?

**Comment by bradleypriest** [comment]
This is using the exact technique I did, however I came to the conclusion it was too naive to work properly. @ferdy-lw did you have any problems with re-rendering deprecation warnings, I havent looked into the source, but I assume the fact the label view is being created before the input view causes the error as it has to redraw the label again once the input and subsequently the elementId is generated for every view. I assume the answer lies in utilizing the runloop or possibly even creating a specific naming convention/counter for the views.

**Comment by ferdy-lw** [comment]
@bradleypriest you're right, I did retest and noticed those re-render warnings - I'm using the latest ember that deprecates the getPath methods so I was just ignoring all the deprecate messages.

I changed the strategy so that when the inputField is inserted it 'sets' the for attribute to it's elementId.  I tried an idea of forcing the inputFields elementId to something known by the label but this seems to be a much worse option and looked like it broke other parts of the binding plumbing.  By having the inputField tell the label it's elementId at least fits in with the proper update binding of attributes after being inserted in the DOM.

I've squashed the commits together so this request only has the 'didInsert' way which differs from the original idea in #24

**Comment by bradleypriest** [comment]
I haven't had a chance to try this yet, but it looks like a better option. A thought though, what do you think about putting the insertElement logic on the wrapper view instead of the input?

File: packages/ember-bootstrap/tests/forms/controls/select_test.js [hunk 1]
```diff
@@ -66,3 +66,10 @@ test("should have the prompt", function() {
   field.set('prompt', 'Select Me');
   equal(select.find('option:eq(0)').text(), "Select Me", "the select has a prompt");
 });
+
+test("should have the label for attribtue", function() {
```
Comment (bradleypriest) [inline packages/ember-bootstrap/tests/forms/controls/select_test.js]: Typo in the test name

File: packages/ember-bootstrap/lib/forms/field.js [hunk 2]
```diff
@@ -26,14 +26,19 @@ Bootstrap.Forms.Field = Ember.View.extend({
       return Bootstrap.Forms.human(value);
     }).property('parentView.label'),
 
-    forBinding: 'value',
+    inputId: 'for',
```
Comment (bradleypriest) [inline packages/ember-bootstrap/lib/forms/field.js]: Maybe `inputElementId` to keep with naming conventions?

**Comment by ferdy-lw** [comment]
I moved the insertElement logic to the wrapper.  This means it's not required on all subclasses of inputField which is much nicer.

While this appears to work it does make me wonder at what point do the browsers use the value of the "for" attribute - if it's checked at DOM insertion time then this technique sets it too late.  The ones I've tried work so I suspect they check the value at the time of clicking.

**Pull request merged**